File: task3.js
let field = [
    [0, 1, 0, 0, 0],
    [0, 0, 1, 0, 0],
    [0, 0, 0, 1, 0],
    [0, 0, 0, 1, 0],
    [0, 0, 1, 0, 0],
];
let safePath = [];

// algorighm
for (let i = 0; i < field.length; i++) {
    let safeX = 0;
    let safeY = 0;
    for (let j = 0; j < field[i].length; j++) {
        if (field[i][j] == 1) {
            if (safePath.length <= 0) {
                safePath.push([i, j]);
                console.log(`Totoshka go [${i},${j}]!`)
            } else {
                let last = safePath[safePath.length - 1];
                console.log(`Ally go [${last[0]},${last[1]}]!`);
                if (j == last[1] || j == last[1] - 1 || j == last[1] + 1) {
                    safePath.push([i, j]);
                    console.log(`Totoshka go [${i},${j}]!`);
                }
            }
        }
    }
}

console.log(`Totoshka is safe!`);
console.log(`Ally is safe!`);

File: Task3.cs
using System;
using System.Collections;

namespace task3
{
    class Program
    {
        static void Main(string[] args)
        {
            const int width = 5;
            const int height = 5;
            int[,] fields = new int[height, width]
                {
                    {0,1,0,0,0},
                    {0,0,1,0,0},
                    {0,0,0,1,0},
                    {0,0,0,1,0},
                    {0,0,1,0,0},
                };

            ArrayList safePath = new ArrayList();
            for (int y = 0; y < height; y++)
            {
                for (int x = 0; x < width; x++)
                {
                    if (fields[y, x] == 1)
                    {
                        if (safePath.Count <= 0)
                        {
                            int[] good = { x, y };
                            safePath.Add(good);
                            Console.WriteLine($"Totoshka go [{y},{x}]");
                        }
                        else
                        {
                            int[] last = (int[])safePath[safePath.Count - 1];
                            Console.WriteLine($"Ally go [{last[1]},{last[0]}]");
                            if (x == last[0] || x == last[0] - 1 || x == last[0] + 1)
                            {
                                int[] good = { x, y };
                                safePath.Add(good);
                                Console.WriteLine($"Totoshka go [{y},{x}]");
                            }


                        }
                    }
                }
            }

            Console.WriteLine($"Totoshka is safe!");
            Console.WriteLine($"Ally is safe!");

        }
    }
}
